### PR TITLE
Update resource types to use hardened images

### PR DESCRIPTION
**Pull request by svenaas**
Addresses cloud-gov/pages-core#4386

## Changes proposed in this pull request:
- Update pipelines to use our current hardened images

## security considerations
Removes slack-notification-resource and replaces with hardened image

### CODE REVIEW

File: ci/pipeline-dev.yml [hunk 1]
```diff
@@ -210,9 +210,13 @@ resources:
...
+    type: registry-image
     source:
-      repository: cfcommunity/slack-notification-resource
+      aws_access_key_id: ((ecr_aws_key))
+      aws_secret_access_key: ((ecr_aws_secret))
+      repository: slack-notification-resource
+      aws_region: us-gov-west-1
+      tag: latest
 
   - name: pull-request
```
Comment (apburnes) [inline ci/pipeline-dev.yml:190]: Missed the dev PR resource

Comment (svenaas) [reply ci/pipeline-dev.yml:190]: So I did. Thank you! Just corrected.

File: ci/pipeline.yml [hunk 2]
```diff
@@ -167,9 +170,10 @@ jobs:
               CF_APP_NAME: pages-builder-((deploy-env))
     on_failure:
       in_parallel:
-        - put: gh-status
-          inputs: [src]
-          params: {state: failure}
+        - put: pr-((git-branch))
+          params:
+            path: pull-request
+            status: failure
```
Comment (apburnes) [inline ci/pipeline.yml]: Add `context: concourse`

**Review by apburnes — COMMENTED** [review]
The tests use `repository: cloudgov/concourse-dcind` lets update that to our recently hardened `pages-dind-v25` repo.

File: ci/pipeline-dev.yml [hunk 3]
```diff
@@ -53,10 +53,13 @@ test: &test
...
       platform: linux
       image_resource:
-        type: docker-image
+        type: registry-image
         source:
-          repository: cloudgov/concourse-dcind
-          tag: original
+          aws_access_key_id: ((ecr-aws-key))
+          aws_secret_access_key: ((ecr-aws-secret))
+          repository: pages-dind-v25
```
Comment (apburnes) [inline ci/pipeline-dev.yml]: It looks like we are using the old `entrypoint.sh` in `ci/docker/` and which fails the task.  We'll want to use [this script](https://github.com/cloud-gov/pages-images/blob/main/dind/v25/dockerd-entrypoint.sh) as the entrypoint and it is already baked into the image. We'll need to rework the test task.

Comment (svenaas) [reply ci/pipeline-dev.yml]: I'm looking at your work on `pages-proxy`.

File: ci/docker/entrypoint.sh [hunk 4]
```diff
@@ -93,12 +93,6 @@ start_docker() {
 
   local docker_opts="${DOCKER_OPTS:-}"
 
-  # Pass through `--garden-mtu` from gardian container
```
Comment (drewbo) [inline ci/docker/entrypoint.sh:96]: what was this doing previously?

**Review by drewbo — APPROVED** [review]
Seems fine on quick review; if it's succeeding on dev builds I'm good

File: ci/docker/entrypoint.sh (hunk 4 above)
Comment (apburnes) [reply ci/docker/entrypoint.sh:96]: Setting the mtu value option.  The hardened container doesn't have iproute2 installed so we can't call `ip route get ...`. The docker default mtu value is 1500 bytes which is far lower than the max so I just removed it.

**Comment by svenaas** [comment]
@apburnes are you good with merging? You've changed this most recently and I don't know whether you've tested builds in dev already.

**Pull request merged**